src/components/primitive/avatar/avatar.component.styles.ts

```diff
@@ -0,0 +1,29 @@
+import styled from 'styled-components';
+import { AvatarComponentProps } from './avatar.component.props';
+import { Image } from '../image/image.component';
+
+/**
+ * Styles for small sized Avatar
+ */
+export const Small = styled(Image)<AvatarComponentProps>`
+  height: 3rem;
+  width: 3rem;
+  border: 1px solid white;
+  border-radius: ${({ flat }) => (flat ? '0.5rem' : '50%')};
+`;
+
+/**
+ * Styles for Medium sized Avatar
+ */
+export const Medium = styled(Small)<AvatarComponentProps>`
+  height: 6rem;
+  width: 6rem;
+`;
+
+/**
+ * Styles for Large sized Avatar
+ */
+export const Large = styled(Medium)<AvatarComponentProps>`
```

Comment:
Why extended the Medium component? You are overriding both the properties either way. You can extend the Small component instead.